backend/src/main/java/kg16/demo/web/LogController.java

```diff
@@ -0,0 +1,81 @@
+package kg16.demo.web;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+class LogController {
+
+    @GetMapping("/logs")
+    public List<Map<String, Object>> getLogs(@RequestParam(required = false) String deviceId,
+                                             @RequestParam(required = false) String startDate,
+                                             @RequestParam(required = false) String endDate,
+                                             @RequestParam(required = false) String alertType) {
+        List<Map<String, Object>> results = new ArrayList<>();
+        String query = "SELECT * FROM OfflineEvents JOIN Scans ON OfflineEvents.device_id = Scans.device_id WHERE 1=1";
+
+        List<Object> params = new ArrayList<>();
+
+        if (deviceId != null) {
+            query += " AND Scans.device_id = ?";
```

Comment:
- Risk for SQLInjection with dynamic query construction `query +=`. Might be ok since its inside preparedStatement but im not sure since im no expert. Maybe safer using stringbuilder for this?